azext_iot/tests/utility/test_iot_file_operations.py

```diff
@@ -0,0 +1,59 @@
+from azext_iot.common.fileops import tar_directory, write_content_to_file
+import pytest
+import os
+from os.path import join, exists
+from azure.cli.core.azclierror import FileOperationError
+
+current_or_empty_dirs = ['./', '.', '', None]
+
+
+class TestFileOperations(object):
+
+    @pytest.mark.parametrize(
+        "content, destination, file_name, overwrite, error, remove_after_test",
+        [
+            # initial create
+            ("test_file_contents", "./", "new_file.txt", False, None, False),
+            # fileoperationerror trying to overwrite existing file with overwrite: false
+            ("new_test_file_contents", "./", "new_file.txt", False, FileOperationError, False),
+            # force overwrite new file
+            ("overwrite_file_contents", "./", "new_file.txt", True, None, True),
+            ("second_test_file_contents", "test_dir", "new_file.txt", True, None, True),
+        ]
+    )
+    def test_write_content_to_file(self, set_cwd, content, destination, file_name, overwrite, error, remove_after_test):
+        try:
+            write_content_to_file(content=content, destination=destination, file_name=file_name, overwrite=overwrite)
+            with open(join(destination, file_name), "rt", encoding="utf-8") as f:
+                assert f.read() == content
+            if remove_after_test:
+                os.remove(join(destination, file_name))
+                if destination not in current_or_empty_dirs:
+                    os.rmdir(destination)
```

Comment:
Add a check that the error did actually get raised and caught in the exception - 
`assert error is None`